upstash/vector/index/api_calls.go

```diff
@@ -0,0 +1,59 @@
+package index
+
+import (
+	"github.com/upstash/terraform-provider-upstash/upstash/client"
+)
+
+func CreateIndex(c *client.UpstashClient, body CreateIndexRequest) (index Index, err error) {
+
+	resp, err := c.SendPostRequest("/v2/vector/index", body, "Create Vector Index", false)
+
+	if err != nil {
+		return index, err
+	}
+
+	err = resp.ToJSON(&index)
+	return index, err
+
+}
+
+func GetIndex(c *client.UpstashClient, indexId string) (index Index, err error) {
+
+	resp, err := c.SendGetRequest("/v2/vector/index/"+indexId, "Get Vector Index", false)
+
+	if err != nil {
+		return index, err
+	}
+
+	err = resp.ToJSON(&index)
+	return index, err
+
+}
+
+func SetIndexPlan(c *client.UpstashClient, indexId string, plan SetPlanRequest) (err error) {
+	_, err = c.SendPostRequest("/v2/vector/index/"+indexId+"/setplan", plan, "Set Plan for Vector Index", false)
+
+	return err
+}
+
+func ResetIndexPassword(c *client.UpstashClient, indexId string) (err error) {
+	_, err = c.SendPostRequest("/v2/vector/index/"+indexId+"/reset-password", nil, "Reset Password for Vector Index", false)
+
+	return err
+}
+
+func TransferIndex(c *client.UpstashClient, indexId string, team string) (err error) {
+	_, err = c.SendPostRequest("/v2/vector/index/"+indexId+"/transfer", team, "Transfer Vector Index to Another Team", false)
+
+	return err
+}
```

Comment:
Transferring for now is a bit risky, there is a chance that user won't be able to manage their tf state after transferring. I would personally want to  remove this functionality unless someone requested it... Wdyt?

Reply:
Is there an update we need to do, to make the transfers possible?